sklearn/tests/test_common.py

```diff
@@ -219,3 +229,124 @@ def test_get_params_invariance():
                     yield check_get_params_invariance, name, Estimator
             else:
                 yield check_get_params_invariance, name, Estimator
+            yield check_transformer_n_iter, name, estimator
+
+
+def test_sample_weight_consistency(random_state=42):
+    estimators = all_estimators()
+
+    n_samples, n_features = 20, 5
+    rng = check_random_state(random_state)
+
+    sample_weight = rng.randint(1, 4, (n_samples,))
+
+    X_clf, y_clf = datasets.make_classification(
+        n_samples=n_samples, n_features=n_features,
+        random_state=random_state)
+    X_reg, y_reg = datasets.make_regression(
+        n_samples=n_samples, n_features=n_features,
+        n_informative=2, random_state=random_state)
+
+    def aug(data, sample_weight):
+        # raise all samples to multiplicity of the corresponding sampleweight
+        aug_data = []
+        for samples, weight in zip(zip(*data), sample_weight):
+            for _ in range(weight):
+                aug_data.append(samples)
+        aug_data = map(np.array, zip(*aug_data))
+        return aug_data
+
+    train, test = train_test_split(range(n_samples))
+
+    for name, Estimator in estimators:
+        if not has_fit_parameter(Estimator, 'sample_weight'):
+            continue
+        if is_classifier(Estimator):
+            X, y = X_clf, y_clf
+        elif is_regressor(Estimator):
+            X, y = X_reg, y_reg
+        else:
+            print ("%s is neither classifier nor regressor" % name)
+            continue
+
+        try:
+            estimator_sw = Estimator()
+            set_random_state(estimator_sw, random_state=random_state)
+            estimator_sw.fit(X[train], y[train], sample_weight=sample_weight[train])
+            X_aug_train, y_aug_train = aug((X[train], y[train]),
+                                       sample_weight[train])
+            estimator_aug = Estimator()
+            set_random_state(estimator_aug, random_state=random_state)
+            estimator_aug.fit(X_aug_train, y_aug_train)
+
+        except ValueError:
```

Comment:
I meant instead of try catch block, it would be safer to have a list of Estimators that have sample_weight in the fit param but might not support it. (which in this case is just LogisticRegression)

wdyt